src/profile/profile.service.ts

```diff
@@ -0,0 +1,382 @@
+     */
+    async getPostsLikedByAccount(
+        accountId: number,
+        limit: number,
+        cursor: string | null,
+    ): Promise<GetProfileDataResult> {
+        const query = this.db('likes')
+            .select(
+                'likes.id as likes_id',
+                // Post fields
+                'posts.id as post_id',
+                'posts.type as post_type',
+                'posts.title as post_title',
+                'posts.excerpt as post_excerpt',
+                'posts.content as post_content',
+                'posts.url as post_url',
+                'posts.image_url as post_image_url',
+                'posts.published_at as post_published_at',
+                'posts.like_count as post_like_count',
+                this.db.raw('1 AS post_liked_by_user'), // Since we are selecting from `likes`, this is always 1
+                'posts.reply_count as post_reply_count',
+                'posts.reading_time_minutes as post_reading_time_minutes',
+                'posts.attachments as post_attachments',
+                'posts.repost_count as post_repost_count',
+                this.db.raw(
+                    `CASE
+                        WHEN reposts.post_id IS NOT NULL THEN 1
+                        ELSE 0
+                    END AS post_reposted_by_user`,
+                ),
+                'posts.ap_id as post_ap_id',
+                // Author fields
+                'author_account.id as author_id',
+                'author_account.name as author_name',
+                'author_account.username as author_username',
+                'author_account.url as author_url',
+                'author_account.avatar_url as author_avatar_url',
+                // Reposter fields
+                'reposter_account.id as reposter_id',
+                'reposter_account.name as reposter_name',
+                'reposter_account.username as reposter_username',
+                'reposter_account.url as reposter_url',
+                'reposter_account.avatar_url as reposter_avatar_url',
+            )
+            .innerJoin('posts', 'posts.id', 'likes.post_id')
+            .innerJoin(
+                'accounts as author_account',
+                'author_account.id',
+                'posts.author_id',
+            )
+            .leftJoin('reposts', function () {
+                this.on('reposts.post_id', 'posts.id').andOnVal(
+                    'reposts.account_id',
+                    '=',
+                    accountId.toString(),
+                );
+            })
+            .leftJoin(
+                'accounts as reposter_account',
+                'reposter_account.id',
+                'reposts.account_id',
+            )
+            .where('likes.account_id', accountId)
+            .modify((query) => {
+                if (cursor) {
+                    query.where('likes.id', '<', cursor);
+                }
+            })
+            .where('posts.deleted_at', null)
```

Comment:
Added this to remove deleted posts from profile but we need a better solution for this. 
When a post is deleted we [remove it from the feed](https://github.com/TryGhost/ActivityPub/blob/main/src/feed/feed-update.service.ts#L87) but it is not removed from likes table and reposts table. This also creates inconsistency while fetching the likesCount for a user.